var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -33,68 +33,100 @@ class Amdfftw(FftwBase):
     variant('openmp', default=True, description="Enable OpenMP support")
-    variant('debug', default=False, description='Builds a debug version of the library')
+    variant('threads', default=False, description="Enable SMP threads support")
+    variant('debug', default=False, description="Builds a debug version of the library")
+    variant(
+        'amd-fast-planner',
+        default=False,
+        description="Option to reduce the planning time without much"
+                    "tradeoff in the performance. It is supported for"
+                    "Float and double precisions only.")
 
     depends_on('texinfo')
 
     provides('fftw-api@3', when='@2:')
 
-    conflicts('precision=quad', when='%aocc', msg="AOCC clang doesn't support quad precision")
-    conflicts('+debug', when='%aocc', msg="AOCC clang doesn't support debug")
+    conflicts('precision=quad', when='@2.2 %aocc', msg="AOCC clang doesn't support quad precision")
+    conflicts('+debug', when='@2.2 %aocc', msg="AOCC clang doesn't support debug")
     conflicts('%gcc@:7.2', when="@2.2:", msg="Required GCC version above 7.2 for AMDFFTW")
+    conflicts('+amd-fast-planner', when="@2.2", msg="amd-fast-planner is supported from 3.0 onwards")
+    conflicts(
+        '+amd-fast-planner',
+        when='precision=quad',
+        msg="amd-fast-planner doesn't support quad precision")
+    conflicts(
+        '+amd-fast-planner',
+        when='precision=long_double',
+        msg="amd-fast-planner doesn't support long_double precision")
 
     def configure(self, spec, prefix):
         """Configure function"""
         # Base options
         options = [
             '--prefix={0}'.format(prefix),
-            '--enable-amd-opt',
-            '--enable-threads'
+            '--enable-amd-opt'
         ]
 
         # Check if compiler is AOCC
         if spec.satisfies('%aocc'):
             options.append("CC={0}".format(os.path.basename(spack_cc)))
-            options.append("CXX={0}".format(os.path.basename(spack_cxx)))
             options.append("FC={0}".format(os.path.basename(spack_fc)))
+            options.append("F77={0}".format(os.path.basename(spack_fc)))
 
         if '+shared' in spec:
             options.append('--enable-shared')
         else:
             options.append('--disable-shared')
 
+        if '+debug' in spec:
+            options.append('--enable-debug')
+
         if '+openmp' in spec:
             options.append('--enable-openmp')
         else:
             options.append('--disable-openmp')
 
+        if '+threads' in spec:
+            options.append('--enable-threads')
+        else:
+            options.append('--disable-threads')
+
         if '+mpi' in spec:
             options.append('--enable-mpi')
             options.append('--enable-amd-mpifft')
         else:
             options.append('--disable-mpi')
             options.append('--disable-amd-mpifft')
 
+        if spec.satisfies('+amd-fast-planner'):
```

Comment:
Is there a reason for the use of `spec.satisfies` here?  References to variants are usually specified as follows:

```suggestion
        if '+amd-fast-planner' in spec:
```

Reply:
Done